pkg/shipper/shipper.go

```diff
@@ -27,11 +28,12 @@ type Remote interface {
 // Shipper watches a directory for matching files and directories and uploads
 // them to a remote data store.
 type Shipper struct {
-	logger log.Logger
-	dir    string
-	remote Remote
-	match  func(os.FileInfo) bool
-	labels func() labels.Labels
+	logger      log.Logger
+	dir         string
+	remote      Remote
+	match       func(os.FileInfo) bool
+	labels      func() labels.Labels
+	metaUpdater cluster.MetadataUpdater
```

Comment:
Looks like the `Shipper` only cares about `SetTimestamps(mint, maxt int64)`.
I'd just continue the pattern and make it a function argument. This way callers (esp. tests) an inject functionality quickly without implementing any particular interface. OTOH, if you have some type with the given method signature, you can still inject it easily.

This also removes the dependency on `pkg/cluster`.

If we used both methods of MetadataUpdater I'd still argue for defining the interface in this package to avoid the dependency.

Reply:
If the function is `nil` a noop should be set similar to `lbls`. Also makes the tests simpler.

Reply:
ok